scenario_09_sandbox.lua

```diff
@@ -38376,151 +38382,236 @@ end
 -- -MAIN FROM KOSAI	F	initialGMFunctions
 -- -CUSTOM				F	customButtons
 -- -ONE-OFFS			F	oneOffs
+snakeCallSigns = {
+	"Żmija", "Wąż", "Gniewosz", "Zaskroniec", "Grzechotnik", "Dusiciel", "Kobra"
+}
+snakeCurrCallSign = 1
 function kosaiOneOff()
 	clearGMFunctions()
 	addGMFunction("-Main From Kosai",initialGMFunctions)
 	addGMFunction("-Custom",customButtons)
 	addGMFunction("-One-Offs",oneOffs)
-	addGMFunction("Exhuari Riptide warp stabilizer outpost", function()
-		-- find the wormhole close to these coordinates
-		-- we do this to separate one-off logic from terrain setup logic
-		local wh = nil
-		local objs = getObjectsInRadius(19791, 114868, 10000)
-		for i=1, #objs do
-			if objs[i].typeName == "WormHole" then
-				wh = objs[i]
-				break
-			end
-		end
-		local originalWhTargetX, originalWhTargetY = wh:getTargetPosition()
-		local whx, why = wh:getPosition()
-		wh:setTargetPosition(whx + irandom(-30000, 30000), why + irandom(-30000, 30000))
+	addGMFunction("Ktlitan Snake", function()
+		removeGMFunction("Ktlitan Snake")
+		addGMFunction(">Ktlitan Snake<", function()
+			onGMClick(nil)
+			kosaiOneOff()
+		end)
+		onGMClick(function(x, y)
+			spawnSnakeFunc(x, y)
+			snakeCurrCallSign = snakeCurrCallSign % #snakeCallSigns + 1
 
+			onGMClick(nil)
+			kosaiOneOff()
+		end)
 
-		local stabilizers = {}
-		local numStabilizers = 5
-		local angleSeparation = 360 / numStabilizers
-		for i=1, numStabilizers do
+		spawnSnakeFunc = function(x, y)
+			local queen = CpuShip():setFaction("Ktlitans"):setTemplate("Ktlitan Queen"):
```

Comment:
There's already a way to create "new" enemy ships (see lines 24248 - 26430 or thereabouts). Among those functions are calls to function addShipToDatabase which puts the details into the science database. You probably did not take advantage of that because you did not see it. In the science database, you get to provide much more than just a description. The description text can be difficult to read because of the tall narrow box in which it appears.

Reply:
Actually this is on purpose. I had intended to add it after the session. After all, this was a previously not encountered ship type.